website/views.py

```diff
@@ -586,6 +586,42 @@ def form_valid(self, form):
     def process_issue(self, user, obj, created, domain, tokenauth=False, score=3):
         p = Points.objects.create(user=user, issue=obj, score=score)
         messages.success(self.request, "Bug added ! +" + str(score))
+        # tweet feature
+        try:
+            auth = tweepy.Client(
+                settings.BEARER_TOKEN,
+                settings.APP_KEY,
+                settings.APP_KEY_SECRET,
+                settings.ACCESS_TOKEN,
+                settings.ACCESS_TOKEN_SECRET,
+            )
+
+            blt_url = "https://" + "%s/issue2/%d" % (
+                settings.DOMAIN_NAME,
+                obj.id,
+            )
+            domain_name = domain.get_name
+            twitter_account = (
+                "@" + domain.get_twitter_account(domain_name) + " "
+                if domain.get_twitter_account(domain_name)
+                else ""
+            )
+
+            issue_title = obj.description + " " if not obj.is_hidden else ""
+
+            message = "%sAn Issue %shas been reported on %s by %s on %s.\n Have look here %s" % (
+                twitter_account,
+                issue_title,
+                domain_name,
+                user.username,
+                settings.PROJECT_NAME,
+                blt_url,
+            )
+
+            auth.create_tweet(text=message)
+            print("\n\n\n\nDone")
+        except Exception as e:
```

Comment:
This catch is too broad, could you narrow it down?

Reply:
@arkid15r I'm not very sure how to do that since there are can be multiple failure such as KEY not setted as well as if api fails there could be other error too. 
could please guide what can i do here ?

Reply:
That's the point -- you need to know/use the context related exceptions instead of catching everything. What's the exception for your key isn't set situation, `AttributeError`, `KeyError`? Same way you should use a specific exception for API failures -- look into available tweepy exceptions and use them. It's better to add missing stuff later than suppress an unexpected error with broad catch.